Core/include/Acts/Propagator/MultiEigenStepperLoop.hpp

```diff
@@ -84,6 +85,18 @@ struct WeightedComponentReducerLoop {
         });
   }
 
+  template <typename stepper_state_t>
+  static Vector3 momentum(const stepper_state_t& s) {
+    return std::accumulate(
+        s.components.begin(), s.components.end(), Vector3::Zero().eval(),
+        [](const auto& sum, const auto& cmp) -> Vector3 {
+          return sum + cmp.weight *
+                           std::abs(cmp.state.absCharge /
```

Comment:
Do we need to account for `cmp.state.absCharge = 0`?

Reply:
charge 0 is kind of broken in all the steppers except the straight line stepper ATM. the future refactor for particle hypothesis is going to improve this hopefully